### PR TITLE
SPVBlockStore: compare magic header as bytes, not string

### PR DESCRIPTION
**Pull request by schildbach**
This gets rid of several bytes to/from string conversions.

Followup of #3431.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/store/SPVBlockStore.java [hunk 1]
```diff
@@ -175,9 +179,8 @@ else if (fileLength < randomAccessFile.length())
     }
 
     private void initNewStore(Block genesisBlock) throws Exception {
-        byte[] header;
-        header = HEADER_MAGIC.getBytes(StandardCharsets.US_ASCII);
-        buffer.put(header);
+        ((Buffer) buffer).rewind();
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/store/SPVBlockStore.java:182]: Isn't this an unrelated change that should be in a separate commit?

**Review by msgilligan — CHANGES_REQUESTED** [review]
I think there is one unrelated change in this PR.

File: core/src/main/java/org/bitcoinj/store/SPVBlockStore.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/store/SPVBlockStore.java]: Indeed. Removed.

**Review by msgilligan — APPROVED** [review]
Looks good!

**Comment by schildbach** [comment]
Merged.
